stdlib/Serialization/src/Serialization.jl

```diff
@@ -748,6 +756,9 @@ function handle_deserialize(s::AbstractSerializer, b::Int32)
         return unwrap_unionall(tname.wrapper)
     elseif b == OBJECT_TAG
         t = deserialize(s)
+        if t === Missing
```

Comment:
`nothing` already has a special representation; in a future version I'll add one for `missing` and then this code can be removed.

Reply:
Agreed, but I would consider that slightly breaking so I'll do it separately.

Reply:
I'm also inclined to say that custom `deserialize` methods aren't called for primitive types (and possibly more things). Of course that's even more breaking, but it's hard to get performance otherwise (e.g. #14678).

Reply:
Hello
The example and the fix seems to be only focusing on vector{union{int, missing}}
How about other primitive types. Eg vector {union {float64,missing}} or float32 float16 bool char?  
Thank you